WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/logic/event/EventStatistics.java

```diff
@@ -0,0 +1,95 @@
+package nl.tudelft.watchdog.intellij.logic.event;
+
+import nl.tudelft.watchdog.core.logic.event.DebugEventManager;
+import nl.tudelft.watchdog.core.logic.event.EventStatisticsBase;
+import nl.tudelft.watchdog.core.logic.event.eventtypes.EventBase;
+import nl.tudelft.watchdog.core.logic.interval.intervaltypes.DebugInterval;
+import org.jfree.data.gantt.GanttCategoryDataset;
+import org.jfree.data.gantt.Task;
+import org.jfree.data.gantt.TaskSeries;
+import org.jfree.data.gantt.TaskSeriesCollection;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Contains the functionality to create a Dataset containing all events that
+ * occurred during a certain debug interval.
+ */
+public class EventStatistics extends EventStatisticsBase {
+
+    /** Constructor. */
+    public EventStatistics(DebugEventManager debugEventManager,
+                           DebugInterval selectedInterval) {
+        super(debugEventManager, selectedInterval);
+    }
+
+    /**
+     * Creates a dataset of all events that occurred during the selected debug
+     * interval.
+     */
+    public GanttCategoryDataset createDebugEventGanttChartDataset() {
+        splitEventsIntoAListPerType();
+
+        // Create and add the tasks for each event type.
+        final TaskSeries allTasks = new TaskSeries("Debug Events");
```

Comment:
That is what I had first, but I had to revert that, see the commit message of 035f5f9b86e3038066f904e6ee230fa676d00322. The problem was that although the building issues can be fixed in IJ and Eclipse itself (although the core project in Eclipse had to depend on org.eclipse.swt, which seems a bit wrong to me), the maven build kept failing as JFreeChart internally depends on SWT and that dependency couldn't be resolved. This is also the reason why those changes you want to unstage are actually there.